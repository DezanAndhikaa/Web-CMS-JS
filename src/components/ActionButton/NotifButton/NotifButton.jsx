import React, { useState } from 'react';
import './NotifButton.scss';
import {Badge, Button, Menu, MenuItem, Popover} from '@material-ui/core/';
// import Popover, {PopoverAnimationVertical} from 'material-ui/core/Popover';
// import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { IconNotif, IconHistory } from '../../../assets/icons';
import { withStyles } from '@material-ui/core/styles';
// import { Menu } from '../../../constants';
import { withRouter } from "react-router-dom";

const Badges = withStyles(theme => ({
	badge: {
		top: -3,
		left: -5,
		fontSize: '10px',
		border: '2px solid white'
	},
}))(Badge);

const DotBadges = withStyles(theme => ({
	badge: {
        top: -1,
		left: 4,
        border: '2px solid white',
        width: 'auto',
        minWidth: '15px',
        height: '15px',
        borderRadius: '50%'
	},
}))(Badge);

class NotifButton extends React.PureComponent {

    constructor(props) {
        super(props);
    
        this.state = {
          displayMenu: false,
        };
      }
    
      showDropdownMenu = (event) => {
        event.preventDefault();
        this.setState({ displayMenu: true }, () => {
          document.addEventListener('click', this.hideDropdownMenu);
        });
      }
    
      hideDropdownMenu = async() => {
        this.setState({ displayMenu: false }, () => {
          document.removeEventListener('click', this.hideDropdownMenu);
        });
        // await this.props.onClickButton();
      }
    
      selectItem = (item) => {
        this.props.onSelectAction(this.props.onSelectActionType, item);
      }
    
      handleClick = (menu, subMenu) => {
          this.props.push(menu);
        }
    
      renderDropdown() {
        return (
            <Button className="notif-history" onClick={this.showDropdownMenu}>
                <DotBadges color="secondary" badgeContent="" anchorOrigin={{ vertical: 'top', horizontal: 'left', }}>
                    <img src={IconHistory} className="icon-notif" alt="" /><span className="label-history">Tracking History</span>
                </DotBadges>
            </Button>
        );
      }
    
      renderDropdownList() {
        return (
          <div className="list-tracking">
              <Button className="list-item-tracking" variant="outlined">
                Sales Order
              </Button>
              <Button className="list-item-tracking" variant="outlined">
               Service Order
              </Button>
          </div>
        );
      }

	render() {
		// if(this.props.titles === "Notif"){
            return(
                <Button className="notif-history">
                    <Badges badgeContent={57} color="secondary" anchorOrigin={{ vertical: 'top', horizontal: 'left', }}>
                        <img src={IconNotif} className="icon-notif" alt="" /> <span className="label-notif">Notification</span>
                    </Badges>
                </Button>
            )
        // } else if (this.props.titles === "History"){
        //     return(
        //         <div className="badan">
        //             {this.renderDropdown()}
        //             {
        //             this.state.displayMenu && this.renderDropdownList()  
        //             }
        //         </div>
        //         <Button className="notif-history" onClick={this.showDropdownMenu}>
        //             <DotBadges color="secondary" badgeContent="" anchorOrigin={{ vertical: 'top', horizontal: 'left', }}>
        //                 <img src={IconHistory} className="icon-notif" alt="" /><span className="label-history">Tracking History</span>
        //             </DotBadges>
        //         </Button>
        //     )
        // } 
	}
}

export default withRouter(NotifButton);