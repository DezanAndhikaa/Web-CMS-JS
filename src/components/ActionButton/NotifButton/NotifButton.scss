@import '../../../themes/variables';
@import '../../../themes/functions';

.dropdown-notif {
  display: flex;
  .dropdown-button-notif {
      border-radius: vw(10);
      display: flex;
      // align-items: center;
      // padding-left: 10%;
      height: vw(45.07);
      .notif-selected-item{            
          margin: 0 0 0 vw(19);
          display: flex;
          flex-direction: row;
          white-space: nowrap;
          align-items: center;
          font-family: 'Lato-Medium';
          font-size: vw(16);
          line-height: vw(17);
          font-weight: 500;
          // overflow: hidden;
          color: black;
          cursor: pointer;
          .expand-icon-container{
              background-image: url('../../../assets/icons/folder.png');
              background-color: map-get($colors, third);
              background-size: 50%;
              background-repeat: no-repeat;
              background-position: center;
              text-align: center;
              position: absolute;
              right: 0;
              height: 100%;
              width: vw(45.07);
              display: flex;
              border-bottom-right-radius: vw(10) !important;
              border-top-right-radius: vw(10) !important;
              .expand-icon{
                  font-size: vw(25);
              }
          }
          .notif-history{
            display: flex;
            flex-direction: row;
            align-self: flex-end;
            cursor: pointer;
            background-color: transparent;
            align-items: center;
            margin-left: 25px;
            .icon-notif{
                // height: 18px; //vw(25) !important
                margin-right: 10px; //vw(10)
            }
            .label-notif{
                font-family: 'Lato-Bold';
                font-size: 14px; //vw(20)
                line-height: 18px; // vw(24)
                color: #3C3B3B;
                // margin-right: 44px; // vw(50)
                text-transform: capitalize;
            }
            .label-notif:hover{
              color: #FC9517;
            }
          }
      }
  }
  .dropdown-button:focus{
      border-style: solid;
      border-color: map-get($colors, third);
  }
}

.list-notif {
  margin: 0;
  right: 0px;
  width: 365px;
  height: 395px;
  // width: 170px; //vw(233.22);
  // height: 110px; // vw(45.07);
  border-radius:vw(10) !important;
  background-color: white;
  position: absolute;
  z-index: 99999;
  box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
  color: rgb(0, 0, 0);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  overflow-y: scroll;
  .list-item-notif {
    padding: vw(6) vw(19);
    color: #3C3B3B;
    font-family: 'Lato-Bold';
    font-size: vw(14);
    line-height: vw(17);
    text-transform: capitalize;
  }
  .list-item-notif:hover {
      background-color: #e5e5e5;
      color: #3C3B3B;
  }
  .see-all-notif{
    position: absolute;
    color: #FFD500;
    font-family: 'Lato-Bold';
    font-size: vw(14);
    line-height: vw(17);
  }
}
.list-notif::before {
  content:"";
  position: absolute;
  right: 40%;
  top: -10px;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 15px 15px 15px;
  border-color: transparent transparent white transparent;
  z-index:0;
}