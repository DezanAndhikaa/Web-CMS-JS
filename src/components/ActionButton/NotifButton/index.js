import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import NotifButton from './NotifButton';

const mapDispatchToProps = (dispatch) => ({
	push : (path) => dispatch(push(path)),
});

const notifbtn = connect(mapDispatchToProps)(NotifButton);

export { notifbtn as NotifButton };