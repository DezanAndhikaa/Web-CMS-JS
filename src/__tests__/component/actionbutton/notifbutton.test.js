import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import NotifButton from "../../../components/ActionButton/NotifButton/NotifButton";

Enzyme.configure({adapter:new Adapter()});
it("render notif button correctly", () => {
  const wrapper = shallow (
    <NotifButton />
  )
  expect(wrapper).toMatchSnapshot();
})