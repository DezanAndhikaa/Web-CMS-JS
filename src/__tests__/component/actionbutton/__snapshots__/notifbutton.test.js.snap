exports[`render notif button correctly 1`] = `ShallowWrapper {}`;